extensions/amp-bind/0.1/bind-impl.js

```diff
@@ -166,16 +166,25 @@ export class Bind {
     /** @type {!Array<./bind-evaluator.BindingDef>} */
     const bindings = [];
 
-    const elements = this.elementsInNode_(body);
-    const numElements = elements.length;
+    const doc = dev().assert(node.ownerDocument, 'ownerDocument is null.');
+    const walker = doc.createTreeWalker(node, NodeFilter.SHOW_ELEMENT);
 
-    // Helper function for scanning a slice of elements.
+    /**
+     * Helper function for scanning a slice of elements.
+     * Returns true if there are no more elements to scan.
+     * @param {number} start
+     * @param {number} end
+     * @return {boolean}
+     */
     const scanFromTo = (start, end) => {
```

Comment:
So it looks like we could simplify this to take an `element`, `boundElements`, and `bindings`, and extract it out of the closure.

```js
function scanElement(element, boundElements, bindings) {
  // basically, the current scanElement_, with a bit of scanFromTo code
}

function chunktion(idleDeadline) {
  let completed = false;
  if (idleDeadline && !idleDeadline.didTimeout) {
    while (idleDeadline.timeRemaining() > 1) {
      var element = walker.next();
      if (!element) {
        completed = true;
        break;
      }
      scanElement(element, boundElements, bindings);
    }
  } else {
    for (let i = 0; i < 250; i++) {
      var element = walker.next();
      if (!element) {
        completed = true;
        break;
      }
      scanElement(element, boundElements, bindings);
    }
  }
}